services/requester/batch_tx_pool.go

```diff
@@ -151,20 +143,46 @@ func (t *BatchTxPool) Add(
 	// For all 3 cases, we record the activity time for the next
 	// transactions that might come from the same EOA.
 	// [X] is equal to the configured `TxBatchInterval` duration.
-	lastActivityTime, found := t.eoaActivity.Get(from)
-	if !found || time.Since(lastActivityTime) > t.config.TxBatchInterval {
-		select {
-		case <-t.done:
-			return fmt.Errorf("the server is shutting down")
-		default:
-			t.txChan <- hexEncodedTx
+	lastActivity, found := t.eoaActivity.Get(from)
+	txHash := tx.Hash()
+
+	// Reject transactions that have already been submitted,
+	// as they are *bound* to fail. Two transactions with
+	// identical hashes, are expected to have the exact same
+	// payload.
+	// - If the 1st was invalid, with a nonce mismatch, the 2nd
+	// is guaranteed to be invalid as well.
+	// - If the 1st was valid, the 2nd is also guaranteed to be
+	// invalid with a nonce mismatch.
```

Comment:
this is not necessarily true. consider the case:
* tx 1 is sent with nonce 5 -> this fails because the nonce should be 4
* tx 2 is sent with nonce 4 -> this succeeds
* tx 1 is sent again with nonce 5 -> this succeeds

Reply:
That's an indicative example of a false negative. I have removed this logic entirely, until we can think of something more solid. My solution was a bit hasty :sweat_smile: 

I kept only the changes to make sending tx synchronous.